test/unit/lib/tools.spec.ts

```diff
@@ -0,0 +1,21 @@
+import { afterAll, beforeAll, describe, expect, test } from 'vitest'
+import { shuffle } from '../../../src/lib/tools'
+
+describe('shuffle', () => {
+    const testArray = [1,2,3,4,5,6]
+
+    test('should return the whole array shuffled', () => {
+        const res = shuffle(testArray)
+
+        expect(res).toHaveLength(testArray.length)
+        expect(res).toEqual(expect.arrayContaining(testArray))
```

Comment:
I feel like this would also work if `shuffle` added elements to the array? At least that's what I'm getting from https://jestjs.io/docs/expect#expectarraycontainingarray, but this whole syntax feels quite unnatural to me.

I guess this is okay thanks to the length check, but what if `testArray` has the same length, skips one number and duplicates another one? (I should probably test this)

Reply:
yeah I agree, this can definitely be improved and I would do that in a different PR :). The main goal here is to set the whole testing environment and then iterate on the tests :)